trade-finance-manager-ui/server/controllers/utilisation-reports/add-payment/index.ts

```diff
@@ -24,25 +51,44 @@ const mapToSelectedReportedFeeViewModel = (feeRecord: SelectedFeeRecordDetails):
   reportedPayments: getFormattedCurrencyAndAmount(feeRecord.reportedPayments),
 });
 
-const mapToAddPaymentViewModel = (data: SelectedFeeRecordsDetails): AddPaymentViewModel => ({
-  bank: data.bank,
-  formattedReportPeriod: getFormattedReportPeriodWithLongMonth(data.reportPeriod),
-  reportedFeeDetails: {
-    totalReportedPayments: getFormattedCurrencyAndAmount(data.totalReportedPayments),
-    feeRecords: data.feeRecords.map((record) => mapToSelectedReportedFeeViewModel(record)),
-  },
+const mapToSelectedReportedFeesDetailsViewModel = (selectedFeeRecordData: SelectedFeeRecordsDetails): SelectedReportedFeesDetailsViewModel => ({
+  totalReportedPayments: getFormattedCurrencyAndAmount(selectedFeeRecordData.totalReportedPayments),
+  feeRecords: selectedFeeRecordData.feeRecords.map((record) => mapToSelectedReportedFeeViewModel(record)),
 });
 
 export const addPayment = async (req: AddPaymentRequest, res: Response) => {
   try {
-    const { userToken } = asUserSession(req.session);
+    const { user, userToken } = asUserSession(req.session);
     const { reportId } = req.params;
     const checkedCheckboxIds = getPremiumPaymentsCheckboxIdsFromObjectKeys(req.body);
     const feeRecordIds = checkedCheckboxIds.map((checkboxId) => getFeeRecordIdFromPremiumPaymentsCheckboxId(checkboxId));
 
-    const data = await api.getSelectedFeeRecordsDetails(Number(reportId), feeRecordIds, userToken);
+    let paymentNumber;
+    let errors: AddPaymentErrorsViewModel = { errorSummary: [] };
+    let formValues: AddPaymentFormValues = { paymentDate: {} };
+    if ('addPaymentFormSubmission' in req.body) {
+      paymentNumber = Number(req.body.paymentNumber);
+      formValues = extractFormValuesFromRequestBody(req.body);
+      errors = validateAddPaymentRequestFormValues(formValues);
+      if (errors.errorSummary.length === 0) {
+        // TODO FN-1739 save data upon valid submission
+        return res.status(501).send('Your request was valid, but we have not finished building the feature');
+      }
+    }
```

Comment:
Could we swap around this order? I think it would be easier to have `if (!('addPaymentFormSubmission' in req.body))` followed by the basic render of the page, then have the rest of the code inside the `if` block to below and use `const` instead of `let`

Reply:
I'm not convinced I prefer it because then we have to have constructing the view model in two places and lots of the lines are the same, so we either make a common function with boatloads of parameters without a particularly nice way to wrap them up, or we have two different places we have to remember to update. What do you think?